File: Templates/Examples/LinqToSql/VBWebApplication/Petshop.UI/PetshopDataService.svc.vb
Imports System.Data.Services
Imports System.Linq
Imports System.ServiceModel.Web

Public Class PetshopDataService
    ' TODO: replace Petshop.Data.PetshopDataContext with your data class name
    Inherits DataService(Of Petshop.Data.PetshopDataContext)

    ' This method is called only once to initialize service-wide policies.
    Public Shared Sub InitializeService(ByVal config As IDataServiceConfiguration)
        ' TODO: set rules to indicate which entity sets and service operations are visible, updatable, etc.
        ' Examples:
        ' config.SetEntitySetAccessRule("MyEntityset", EntitySetRights.AllRead)
        ' config.SetServiceOperationAccessRule("MyServiceOperation", ServiceOperationRights.All)
    End Sub

End Class

File: Projects/CSharp/CSharpWebApplication/Petshop.UI/PetshopDataService.svc.cs
using System;
using System.Collections.Generic;
using System.Data.Services;
using System.Linq;
using System.ServiceModel.Web;
using System.Web;

namespace Petshop.UI
{
    public class PetshopDataService : DataService<Petshop.Data.PetshopDataContext>
    {
        // This method is called only once to initialize service-wide policies.
        public static void InitializeService(IDataServiceConfiguration config)
        {
            // TODO: set rules to indicate which entity sets and service operations are visible, updatable, etc.
            // Examples:
            // config.SetEntitySetAccessRule("MyEntityset", EntitySetRights.AllRead);
            // config.SetServiceOperationAccessRule("MyServiceOperation", ServiceOperationRights.All);
        }
    }
}
